docs/decisions/app/0005-server-rendering.md

```diff
@@ -0,0 +1,73 @@
+## Context and Problem Statement
+
+Next.js provides multiple ways of rendering a website, which have different commands: `next start` vs `next export`. For the platform, we need to determine which should be the default.
+
+## Considered options
+
+1.  Server rendering (`next start`): Generates the full HTML markup for a page on the server in response to navigation. Minimal client-side JS is possible, even for personalized sites.
+2.  Prerendering (`next export`): At compile time, a separate HTML file is generated for each URL. Only the initial state of the page is generated as static HTML. To display personalized data, client-side JS is required and the page's DOM is updated.
+
+## Decision drivers
+
+1.  The selected option should represent what we consider the preferred approach for the types of websites we're typically building with Next.js: authenticated, personalized web applications (claimant portals, compare tools, case management systems, etc).
+2.  Reduce the need for third party dependencies or custom code, when a native option works just as well.
+3.  A (reasonable) increase in cloud costs is acceptable if it results in a system that is more maintainable for software teams and the government in the long term.
+4.  Developer experience should not come at the expense of end user experience.
```

Comment:
Can we restate this in less black and white terms? e.g. something like "Prioritize end user experience above developer experience" To quote [the article I saw you shared](https://seldo.com/posts/the_case_for_frameworks):

> But on the other hand, if the website were designed to run perfectly for every single customer, in every location, on every device, then the developers would spend their entire time in optimization work and never ship anything new.

As an easy strawman example to pick on, I think we have generally agreed that we're not going to spend to much effort on supporting old versions of internet explorer.